Allow Listening to Specific Hosts

lkiesow opened the pull request:
This patch allows to listen to a specific host only.

For example, when BigBlueButton is used behind Nginx as reverse Proxy,
it is completely sufficient to listen to 127.0.0.1 only, but right now
there is now easy way to configure that.

Note that this patch does not change the default and still works with
the previous configuration making updates easy.

prlanzarin commented:
@lkiesow thanks for the PR. Agree with the premise, it was an oversight. I'll review and test it.

lkiesow commented:
For the packages, this should probably even be configured to `127.0.0.1` since these are running behind Nginx as reverse proxy anyway. Not sure if you are involved in the packaging though.

prlanzarin commented:
@lkiesow most of the default configs are piped through the packaging stuff, so it'll probably stay as localhost as in your PR.

By the way, we also require a signed Contributor License Agreement from people making contributions to BBB and it's related components. See http://docs.bigbluebutton.org/support/faq.html#why-do-i-need-to-sign-a-contributor-license-agreement-to-contribute-source-code.

lkiesow commented:
Hi @prlanzarin, thanks for the review. I've sent a CLA to Fred though he seems to be pretty under right now. I've also added two commits incorporating your suggestions:

- Added environment variables to control the new settings
- Made listening to `127.0.0.1` the default as it should be sufficient in most cases (especially the default packaging) and usually goes into packaging unmodified if I understood you correctly

lkiesow commented:
Hi @prlanzarin, is anything else missing here?
